R/load-stats19.R

```diff
@@ -0,0 +1,251 @@
+#' Download Stats19 data
+#'
+#' @section Details:
+#' This convenience function downloads and unzips UK road traffic casualty data.
+#' It results in unzipped .csv data in R's temporary directory.
+#'
+#' Ensure you have a fast internet connection and at least 100 Mb space
+#'
+#' @param zip_url The url where the data is stored
+#' @export
+#' @examples
+#' \dontrun{
+#' dl_stats19()
+#' }
+dl_stats19 <- function(zip_url = paste0("http://data.dft.gov.uk.s3.amazonaws.com/",
+  "road-accidents-safety-data/Stats19_Data_2005-2014.zip")){
+
+  # download and unzip the data if it's not present
+  if(!"Accidents0514.csv" %in% list.files(tempdir())){
+    data_dir <- tempdir()
+    destfile <- file.path(data_dir, "Stats19_Data_2005-2014.zip")
+    downloader::download(zip_url, destfile)
+    unzip(destfile, exdir = data_dir)
+  }
+
+  list.files(tempdir())
+  print(paste0("Data saved at: ", list.files(tempdir(),
+              pattern = "csv", full.names = TRUE)))
+
+}
+
+#' Import and format UK 'Stats19' road traffic casualty data
+#'
+#' @section Details:
+#' This is a wrapper function to access and load stats 19 data in a user-friendly way.
+#' The function returns a data frame, in which each record is a reported incident in the
+#' stats19 dataset.
+#'
+#' Ensure you have a fast internet connection and at least 100 Mb space.
+#'
+#' @param data_dir Character string representing where the data is stored.
+#' If empty, R will attempt to download and unzip the data for you.
+#' @export
+#' @examples
+#' \dontrun{
+#' ac <- read_stats19_ac()
+#' }
+read_stats19_ac <- function(data_dir = tempdir()){
+  if(!"Accidents0514.csv" %in% list.files(data_dir)){
```

Comment:
I think it is best not to hardcode file names. We never know if somebody will want to make some modifications to the file and then rename it "Accidents0514_fixed.csv". Can we change this to add a second parameter for the filename with "Accidents0514.csv" as the default value?